db-async-common/src/main/scala/com/github/mauricio/async/db/pool/SingleThreadedAsyncObjectPool.scala

```diff
@@ -49,9 +49,9 @@ class SingleThreadedAsyncObjectPool[T](
   import SingleThreadedAsyncObjectPool.{Counter, log}
 
   private val mainPool = Worker()
-  private val poolables = new ArrayBuffer[PoolableHolder[T]](configuration.maxObjects)
+  private var poolables = new Stack[PoolableHolder[T]]()
```

Comment:
because of line 244: `this.poolables = this.poolables.diff(removals)`. Unfortunately mutable.Stack doesn't support removing elements that are not the top itiem, as far as I understand.